Support JDK11 in generated maven project

The generated maven project can't be compiled with JDK 11. As the next Version of AEM will also work with JDK 11 this archetype should also support this JDK version.

current output

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:compile (default-compile) on project de.bpauli.test.core: Compilation failure: Compilation failure:
[ERROR] /Users/pauli/Dev/CQ/Tests/de.bpauli.test/core/src/main/java/de/bpauli/core/models/HelloWorldModel.java:[18,24] package javax.annotation does not exist
[ERROR] /Users/pauli/Dev/CQ/Tests/de.bpauli.test/core/src/main/java/de/bpauli/core/models/HelloWorldModel.java:[39,6] cannot find symbol
[ERROR]   symbol:   class PostConstruct
[ERROR]   location: class de.bpauli.core.models.HelloWorldModel
[ERROR] -> [Help 1]
```

CQ-4268553

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -23,40 +23,40 @@ version: 2
 shared: &shared
   working_directory: ~/repo
 
-    environment:
-      # Customize the JVM maximum heap limit
-      MAVEN_OPTS: -Xmx3200m
+  environment:
+    # Customize the JVM maximum heap limit
+    MAVEN_OPTS: -Xmx3200m
 
-    steps:
-      - checkout
+  steps:
+    - checkout
 
-      # Download and cache dependencies
-      - restore_cache:
-          keys:
-            - v1-dependencies-{{ checksum "pom.xml" }}
-            # fallback to using the latest cache if no exact match is found
-            - v1-dependencies-
+    # Download and cache dependencies
+    - restore_cache:
+        keys:
+          - v1-dependencies-{{ checksum "pom.xml" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
 
-      - run: mvn dependency:go-offline
+    - run: mvn dependency:go-offline
 
-      - save_cache:
-          paths:
-            - ~/.m2
-          key: v1-dependencies-{{ checksum "pom.xml" }}
+    - save_cache:
+        paths:
+          - ~/.m2
+        key: v1-dependencies-{{ checksum "pom.xml" }}
 
-      # run tests!
-      - run: mvn integration-test
+    # run tests!
+    - run: mvn integration-test
 
-      - run:
-          name: Save test results
-          command: |
-            mkdir -p ~/test-results/junit/
-            find . -type f -regex ".*/surefire-reports/.*" -exec cp {} ~/test-results/junit/ \;
-          when: always
-      - store_test_results:
-          path: ~/test-results
-      - store_artifacts:
-          path: ~/test-results/junit
+    - run:
+        name: Save test results
+        command: |
+          mkdir -p ~/test-results/junit/
+          find . -type f -regex ".*/surefire-reports/.*" -exec cp {} ~/test-results/junit/ \;
+        when: always
+    - store_test_results:
+        path: ~/test-results
+    - store_artifacts:
+        path: ~/test-results/junit
 
 jobs:
   java-8:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -59,7 +59,7 @@ shared: &shared
         path: ~/test-results/junit
 
 jobs:
-  java-8:
+  build:
     docker:
       - image: circleci/openjdk:8-jdk
     <<: *shared

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -59,12 +59,19 @@ shared: &shared
         path: ~/test-results/junit
 
 jobs:
-  build:
+  build-java-8:
     docker:
       - image: circleci/openjdk:8-jdk
     <<: *shared
 
-  java-11:
+  build-java-11:
     docker:
       - image: circleci/openjdk:11-jdk
     <<: *shared
+
+workflows:
+  version: 2
+  build:
+    jobs:
+      - build-java-8
+      - build-java-11

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -19,18 +19,9 @@
 # Check https://circleci.com/docs/2.0/language-java/ for more details
 #
 version: 2
-jobs:
-  build:
-    docker:
-      # specify the version you desire here
-      - image: circleci/openjdk:8-jdk
 
-      # Specify service dependencies here if necessary
-      # CircleCI maintains a library of pre-built images
-      # documented at https://circleci.com/docs/2.0/circleci-images/
-      # - image: circleci/postgres:9.4
-
-    working_directory: ~/repo
+shared: &shared
+  working_directory: ~/repo
 
     environment:
       # Customize the JVM maximum heap limit
@@ -67,3 +58,13 @@ jobs:
       - store_artifacts:
           path: ~/test-results/junit
 
+jobs:
+  java-8:
+    docker:
+      - image: circleci/openjdk:8-jdk
+    <<: *shared
+
+  java-11:
+    docker:
+      - image: circleci/openjdk:11-jdk
+    <<: *shared

--- a/src/main/archetype/core/pom.xml
+++ b/src/main/archetype/core/pom.xml
@@ -58,25 +58,35 @@
                 <configuration>
                     <exportScr>true</exportScr>
                     <instructions>
-                        <!-- Import any version of javax.inject, to allow 
-                            running on multiple versions of AEM -->
-                        <Import-Package>javax.inject;version=0.0.0,*</Import-Package>
+                        <!-- Import any version of javax.inject and javax.annotation, to allow running on multiple versions of AEM -->
+                        <Import-Package>
+                            javax.inject;version=0.0.0,
+                            javax.annotation;version=0.0.0,
+                            *
+                        </Import-Package>
                         <Sling-Model-Packages>
                             ${package}.core
                         </Sling-Model-Packages>
                         <_dsannotations>*</_dsannotations>
                         <_metatypeannotations>*</_metatypeannotations>
-                        <!-- Allow the processing of SCR annotations via 
-                            a bnd plugin -->
-                        <_plugin>org.apache.felix.scrplugin.bnd.SCRDescriptorBndPlugin;destdir=${project.build.outputDirectory}</_plugin>
+                        <_plugin>
+                            <!-- Enable registration of Sling Models classes via bnd plugin -->
+                            org.apache.sling.bnd.models.ModelsScannerPlugin,
+                            <!-- Allow the processing of SCR annotations via a bnd plugin -->
+                            org.apache.felix.scrplugin.bnd.SCRDescriptorBndPlugin;destdir=${project.build.outputDirectory}
+                        </_plugin>
                     </instructions>
                 </configuration>
                 <dependencies>
-                    <!-- Add a dependency to the bnd plugin -->
                     <dependency>
                         <groupId>org.apache.felix</groupId>
                         <artifactId>org.apache.felix.scr.bnd</artifactId>
                         <version>1.9.0</version>
+                    </dependency>
+                    <dependency>
+                        <groupId>org.apache.sling</groupId>
+                        <artifactId>org.apache.sling.bnd.models</artifactId>
+                        <version>1.0.0</version>
                     </dependency>
                 </dependencies>
             </plugin>
@@ -104,6 +114,10 @@
         <dependency>
             <groupId>javax.inject</groupId>
             <artifactId>javax.inject</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>javax.annotation</groupId>
+            <artifactId>javax.annotation-api</artifactId>
         </dependency>
         <!-- Other Dependencies -->
         <dependency>

--- a/src/main/archetype/pom.xml
+++ b/src/main/archetype/pom.xml
@@ -513,6 +513,12 @@
                 <artifactId>javax.inject</artifactId>
                 <version>1</version>
             </dependency>
+            <dependency>
+                <groupId>javax.annotation</groupId>
+                <artifactId>javax.annotation-api</artifactId>
+                <version>1.3.2</version>
+                <scope>provided</scope>
+            </dependency>
             <!-- Logging Dependencies -->
             <dependency>
                 <groupId>org.slf4j</groupId>

